src/utils/fileReader/index.ts

```diff
@@ -34,22 +39,31 @@ const fileReaderFactory = (options: Partial<FileReaderOptions>) => {
           );
         }
 
-        const { name, type, size, lastModified } = file;
+        const { type, size } = file;
+        let name: File['name'] | undefined;
+        let lastModified: File['lastModified'] | undefined;
+
+        if (file instanceof File) {
```

Comment:
fileReader can also handle blobs, and in the case of thumbnail optimisation, a `Blob` is returned by the optimiser. Blobs don't have `name` or `lastModified` fields, hence adding this check.